src/api/getWilayaByDairaName.js

```diff
@@ -0,0 +1,27 @@
+const projectWilaya = require('../utils/projections/wilayaProjection');
+
+const getWilayaByDairaName = data =>
+/**
+ * Takes a wilaya daira-name and returns its respective wilaya
+ *
+ * @example <p> Get Relizane wilaya by 'OUED RHIOU' daira
+ * //returns {name: "Relizane", ...}
+ * getWilayaByDairaName("")
+ *
+ * @param { String } daira The Wilaya's daira name(en | ar | fr)
+ * @param {String[]} projection a list of  wilaya object attributes to keep
+ * @returns { Object | undefined } Returns matching wilaya, or undefined
+ */
+  (daira, projection) =>  {
```

Comment:
Please add a check if `daira` is not empty or not null (`!daira || daira.tim().length`). If so return `undefined` (you can disable eslint warning for this line as it will pop-up).

Also, please do a case incentive comparison, _it may make sense to do something like this:_

```js
  (daira, projection) =>  {
  const dairaName = daira.trim().toLowerCase()
```